.github/workflows/ci.yml

```diff
@@ -1,17 +1,21 @@
 name: CI
 
-on: [pull_request]
+on: [push, pull_request]
 
 jobs:
   build:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node-version }}
       - run: yarn install --frozen-lockfile
       - run: yarn test
       - run: yarn build
+      - run: yarn publish-canary
+        if: ${{ github.ref == 'refs/heads/main' }}
```

Comment:
instead of having this `if` gate, my first instinct was that we should break this into `ci.yml` and `cd.yml`, but after thinking about it, I actually think this is better. I'm on board with having it all in `ci.yml` for consistency 👍🏼

Reply:
we can still change when adding the manual NPM release work flow. I'm fine with either